server/api/externalapi.ts

```diff
@@ -32,13 +32,28 @@ class ExternalAPI {
       this.fetch = fetch;
     }
 
-    this.baseUrl = baseUrl;
-    this.params = params;
+    const url = new URL(baseUrl);
+
     this.defaultHeaders = {
       'Content-Type': 'application/json',
       Accept: 'application/json',
+      Authorization:
+        url.username || url.password
+          ? `Basic ${Buffer.from(`${url.username}:${url.password}`).toString(
+              'base64'
+            )}`
+          : undefined,
```

Comment:
```suggestion
const url = new URL(baseUrl);
this.defaultHeaders = {
  'Content-Type': 'application/json',
  Accept: 'application/json',
  ...(url.username || url.password) && {
    Authorization: `Basic ${Buffer.from(`${url.username || ''}:${url.password || ''}`).toString('base64')}`
  },
  ...options.headers,
};
```
Wouldn't it be better to pass the authorisation header conditionally as opposed to passing an empty string (undefined won't work here).

It doesn't include an empty Authorization header when there are no credentials. It's more semantically correct as there are no Authorization header at all vs an empty one. And it's cleaner in terms of HTTP headers being sent.

@gauthier-th